plugins/catalog-react/src/hooks/useEntityListProvider.tsx

```diff
@@ -116,6 +118,16 @@ export const EntityListProvider = <EntityFilters extends DefaultEntityFilters>({
       (allQueryParams.filters as Record<string, string | string[]>) ?? {},
   });
 
+  // Use useEffect cleanup mechanism to flag when component is unmounted
+  // so that we can cleanup any residual async callbacks
+  const unmounted = useRef(false);
```

Comment:
react-use (which we use a lot) has you covered!

https://github.com/streamich/react-use/blob/master/docs/useMountedState.md

Reply:
This is probably something thah we should use a LOT more than we do around the code base.

Reply:
ah, thanks for pointing out this nifty hook!